dxp/sennajs_integration.md

```diff
@@ -0,0 +1,117 @@
+Additionally, if you want to enable Single Application Mode for Internet Explorer, you can do so by navigating to: **Control Panel** > **System Settings** > **Infrastructure** > **Frontend SPA Infrastructure** and unchecking the checkboxes for Internet Explorer, but keep in mind that this might change in the future and that it is not recommended.
+
+## Link selectors
+
+By default, the Senna.js `App` instance will handle all click events on links (`<a>` tags), that match the CSS selectors specified in the `Liferay.SPA.navigationExceptionSelectors` global variable.
+
+These CSS selectors can also be configured by navigating to **Control Panel** > **System Settings** > **Infrastructure** > **Frontend SPA Infrastructure** and changing the values specificed in the **Navigation Exception Selectors** fields.
+
+## Form submission
+
+The form submission will also be handled by the Senna.js `App` instance, unless the form's `action` attribute is set to `get`
+
+## Block list
+
+It's also possible to tell the Senna.js `App` instance which portlets it needs to ignore, in which case it will not be used for navigation or form submission.
+
+Any portlet that is set to not use "Single Application Mode", or any portlet that is not deployed (and not ready) is automatically added to this list.
+
+## Cache
+
+The Senna.js `App` will use caching, if the `Liferay.SPA.cacheExpirationTime` variable's value is greater than `-1`.
+By default DXP sets this value to `-1` which means that the cache is disabled, if you want Senna.js cache to never be cleared you can set this value to `0`
+or specify the value (in minutes) after when Senna.js cache should be cleared.
+
+The cache expiration time can also be configured by navigating to **Control Panel** > **System Settings** > **Infrastructure** > **Frontend SPA Infrastructure** and changing the value specified in the **Cache Expiration Time** field.
+
+With caching enabled, once a resource is loaded, it's content is cached into memory and is retrieved later on without any additional request.
+
+## Excluded paths
+
+The `frontend-js-spa-web` module also introduces a property called [`spa.exluced.paths`](https://github.com/liferay/liferay-portal/blob/454d7f6a439dc668a0ce0529cff2cae87ee9a0d5/modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/spa.properties) which, as it's name indicates, is a list of paths that should be ignored by the `Senna.js` App instance.
+
+These paths, are visible in the global `Liferay.SPA.excludedPaths` variable.
+
+## URLs
+
+By default Senna.js in DXP will contain the required parameters needed by portlets, specifically `p_p_id`, `p_p_lifecycle`, `p_p_state` and `p_p_auth`
+
+## How Senna.js works
+
+![Simplified overview of Senna.js](../images/sennajs-overview.png)
+
+Senna.js takes into account concepts for dealing with SPA on sites that are not built purely on a framework/library, such as React, Vue or Angular.
+
+Senna.js, introduces the following components:
```

Comment:
Can we call them "concepts" instead of "components"? (Otherwise they sound a bit like "UI components")

Reply:
Yes of course, I just didn't come up with a better name - but it's true that "components" is confusing in this case.